x-pack/metricbeat/module/googlecloud/stackdriver/metadata_services.go

```diff
@@ -17,6 +17,12 @@ func NewMetadataServiceForConfig(c config) (googlecloud.MetadataService, error)
 	switch c.ServiceName {
 	case googlecloud.ServiceCompute:
 		return compute.NewMetadataService(c.ProjectID, c.Zone, c.opt...)
+	case googlecloud.ServicePubsub:
+		return nil, nil
+	case googlecloud.ServiceLoadBalancing:
+		return nil, nil
+	case googlecloud.ServiceStorage:
```

Comment:
It seems that `ServicePubsub` and `ServiceLoadBalancing` are not strictly related to this patch and related to different upcoming metricsets, right?

Also `ServiceStorage` returns nil metadata too?

If we plan to keep it like this for now could you please provide some content here for reference so as to be more clear?

Reply:
Yeah, main thing here is that I'm trying to avoid too many potential conflicts because there are 3 different follow-up PR's touching the same function and I have to wait until only one is left to see overall final look.

About the question of storage: This file, `metadata_services.go` "controls / knows" which metricsets / gcp-services have an specific implementation to retrieve extra metadata (call it labels or properties) using their specific API's, instead of taking just the labels that are returned in the Stackdriver API. Only one metricset has such implementation: [`compute`](https://github.com/elastic/beats/tree/master/x-pack/metricbeat/module/googlecloud/stackdriver/compute).

This is caused by a direct limitation on lightweight modules. Being _so_ generic is a con in this case.

Reply:
Fair enough! Thanks for elaborating!